### DISCUSSION TITLE
[Feature] [Transcoding] Always treat H264 as desired formats (and other plans)

### DISCUSSION THREAD

**Discussion by Hely0n** [immich-app/immich#6319] [Feature Request]
### The feature

Hello everyone,

I would like to implement some changes to the transcode policies and this would be the first one, I want to work on:

As already discussed in the discord, H264 seems to run everywhere, so there is no need to transcode it for compatiblity reasons.
My first implementation would be that H264 will be treated as a desired format in addition to the one the user selected.

After I successfully implemented the basic feature, I would try to extend it to a "desired formats" selection in the UI, so the user can check which codecs and containers the user "desires", but I can't promise anything here, as I'm currently not that familiar with the UI code of immich and Svelte. But I will try!
 
My next planned feature would be to add a transcode policy called "Not in desired format or higher than target max bitrate". Here the admin could set a bitrate, that matches the upload speed of the server connection for example, so only videos will be transcoded that exceed given bitrate independently of their resoultion.
This would also cover the needs of users who wanted to have H264-Videos transcoded just for the lower bandwidth. 

Please feel free to give some feedback about the ideas. I'm open to changes to the plans 😁

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by aisbergde** [comment]
And together with implementing this, there could be an option to clean up / remove previously transcoded content, depending on the selected new policy.

BTW, is there an option, to disable transcoding right now, until this is implemented?

**Reply by Hely0n** [reply]
Since the last update, transcoded videos which does not meet the Policy, will be removed when you start "all" on the jobs page. But not sure if it's also done of you click "missing"

And you can disable transcoding by selecting the appropriate transcoding policy in the admin settings under "video transcoding"

**Reply by mertalev** [reply]
Missing won't delete anything since it will only queue assets that don't have transcodes.

**Reply by aisbergde** [reply]
idea for an implementation (or maybe just a database query)
- remove not needed transcoded content, depending on settings

**Reply by mertalev** [reply]
It could maybe be added as a nightly job.

**Reply by mertalev** [reply]
Actually, it'd be better as something that runs after a change in transcode policy.
